✨ Describe capd and add goals in readme (#255)
kubernetes-sigs/cluster-api-provider-docker: README.md

# Cluster API Provider Docker (CAPD)

> CAPD is a reference implementation of an infrastructure provider for Cluster API project using Docker.

Reviewer: ```suggestion
CAPD is a reference implementation of an infrastructure provider for the Cluster API project using Docker.
```

This is one out of three components needed to run a Cluster API management cluster.

> For a complete overview, please refer to documentation available [here](https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm#cluster-api-bootstrap-provider-kubeadm) which uses CAPD as an example infrastructure provider.

Reviewer: ```suggestion
For a complete overview, please refer to the documentation available [here](https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm#cluster-api-bootstrap-provider-kubeadm) which uses CAPD as an example infrastructure provider.
```

## CAPD Goals

> * To become the reference implementation of an infrastructure provider.

Reviewer: ```suggestion
* To be a reference implementation of an infrastructure provider.
```

> * Being a reference implemnetation means that the code is highly trusted and can be used in testing.

Reviewer: ```suggestion
* The code is highly trusted and used in testing of Cluster API
```

> * This provider can be used as a guide for develipers looking to implement their own infrastructure provider.

Reviewer: ```suggestion
* This provider can be used as a guide for developers looking to implement their own infrastructure provider.
```

Author: Thank you for tolerating my sloppiness :)

Reviewer: Thank you for taking the time to submit docs PRs :)

> * Providers looking to implement their own infrastructure controller will be directed to this implementation if there is a question about the Cluster API specification/contracts.

Reviewer: ```suggestion
* This provider can be used as a guide for developers looking to implement their own infrastructure provider.
```
